### ISSUE TITLE
LogStash::Runner Specs are Failing!

### ISSUE THREAD

**Issue by original-brownbear** [elastic/logstash#7460]
See here:

https://travis-ci.org/elastic/logstash/jobs/243074345#L5541

```sh
rspec ./logstash-core/spec/logstash/runner_spec.rb:350 # LogStash::Runner logstash modules --modules with an available module specified but no connection to elasticsearch should log fatally and return a bad exit code
rspec ./logstash-core/spec/logstash/runner_spec.rb:380 # LogStash::Runner logstash modules --modules with an available module specified and a mocked connection to elasticsearch should not terminate logstash
rspec ./logstash-core/spec/logstash/runner_spec.rb:317 # LogStash::Runner logstash modules --config.test_and_exit with a good configuration should exit successfully
```

I can reproduce this in 100% of runs locally as well, relevant output there:

```sh
ERROR: Configuration reloading also requires passing a configuration path with '-f yourlogstash.conf'
WARNING: Could not find logstash.yml which is typically located in $LS_HOME/config or /etc/logstash. You can specify the path using --path.settings. Continuing using the defaults
WARNING: Could not find logstash.yml which is typically located in $LS_HOME/config or /etc/logstash. You can specify the path using --path.settings. Continuing using the defaults
2017-06-15 11:51:17,271 LogStash::Runner ERROR No log4j2 configuration file found. Using default configuration: logging only errors to the console.
Configuration OK
2017-06-15 11:51:22,508 LogStash::Runner ERROR No log4j2 configuration file found. Using default configuration: logging only errors to the console.
ERROR: option '-w': Setting "pipeline.workers" must be a Integer. Received:  (NilClass)

See: 'bin/logstash --help'
2017-06-15 11:51:29,042 LogStash::Runner ERROR No log4j2 configuration file found. Using default configuration: logging only errors to the console.

RSpec::Mocks::MockExpectationError: (Double "logger").fatal("The given configuration is invalid. Reason: Could not load the configuration file")
    expected: 0 times with any arguments
    received: 1 time with arguments: ("The given configuration is invalid. Reason: Could not load the configuration file")
./logstash-core/lib/logstash/runner.rb:316:in `execute'
./vendor/bundle/jruby/2.3.0/gems/clamp-0.6.5/lib/clamp/command.rb:67:in `run'
./logstash-core/lib/logstash/runner.rb:229:in `run'
./logstash-core/spec/logstash/runner_spec.rb:319:in `block in (root)'
./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
./lib/bootstrap/rspec.rb:13:in `<main>'
-e:1:in `<main>'
------------------
--- Caused by: ---
RuntimeError:
  Could not load the configuration file
  ./logstash-core/lib/logstash/runner.rb:312:in `execute'

RSpec::Mocks::MockExpectationError: (Double "logger").fatal("An unexpected error occurred!", {:error=>#<LogStash::ConfigLoadingError: The modules specified are not available yet. Specified modul.../bootstrap/rspec.rb:13:in `<main>'", "org/jruby/RubyKernel.java:979:in `load'", "-e:1:in `<main>'"]})
    expected: 0 times with any arguments
    received: 1 time with arguments: ("An unexpected error occurred!", {:error=>#<LogStash::ConfigLoadingError: The modules specified are not available yet. Specified modul.../bootstrap/rspec.rb:13:in `<main>'", "org/jruby/RubyKernel.java:979:in `load'", "-e:1:in `<main>'"]})
./logstash-core/lib/logstash/runner.rb:360:in `execute'
./vendor/bundle/jruby/2.3.0/gems/clamp-0.6.5/lib/clamp/command.rb:67:in `run'
./logstash-core/lib/logstash/runner.rb:229:in `run'
./vendor/bundle/jruby/2.3.0/gems/clamp-0.6.5/lib/clamp/command.rb:132:in `run'
./logstash-core/spec/logstash/runner_spec.rb:381:in `block in (root)'
./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
./lib/bootstrap/rspec.rb:13:in `<main>'
-e:1:in `<main>'
------------------
--- Caused by: ---
LogStash::ConfigLoadingError:
  The modules specified are not available yet. Specified modules: ["cef"] Available modules: ["fb_apache"]
  ./logstash-core/lib/logstash/config/modules_common.rb:33:in `pipeline_configs'

expected {:error=>#<LogStash::ConfigLoadingError: The modules specified are not available yet. Specified modul.../bootstrap/rspec.rb:13:in `<main>'", "org/jruby/RubyKernel.java:979:in `load'", "-e:1:in `<main>'"]} to be a config loading error hash /Failed to import module configurations to Elasticsearch. Module: cef/
./logstash-core/spec/logstash/runner_spec.rb:341:in `block in (root)'
./logstash-core/lib/logstash/runner.rb:360:in `execute'
./vendor/bundle/jruby/2.3.0/gems/clamp-0.6.5/lib/clamp/command.rb:67:in `run'
./logstash-core/lib/logstash/runner.rb:229:in `run'
./vendor/bundle/jruby/2.3.0/gems/clamp-0.6.5/lib/clamp/command.rb:132:in `run'
./logstash-core/spec/logstash/runner_spec.rb:351:in `block in (root)'
./vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in /Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb'
./lib/bootstrap/rspec.rb:13:in `<main>'
-e:1:in `<main>'
------------------
--- Caused by: ---
LogStash::ConfigLoadingError:
  The modules specified are not available yet. Specified modules: ["cef"] Available modules: ["fb_apache"]
  ./logstash-core/lib/logstash/config/modules_common.rb:33:in `pipeline_configs'
/Users/brownbear/src/logstash/vendor/bundle/jruby/2.3.0/gems/rspec-mocks-3.6.0/lib/rspec/mocks/instance_method_stasher.rb:8: warning: singleton on non-persistent Java type Java::OrgLogstashCommonIo::DeadLetterQueueWriter (http://wiki.jruby.org/Persistence)
11:53:15.985 [Ruby-0-Thread-646: /Users/brownbear/src/logstash/logstash-core/spec/logstash/shutdown_watcher_spec.rb:61] ERROR logstash.shutdownwatcher - The shutdown process appears to be stalled due to busy or blocked plugins. Check the logs for more information.
11:53:17.065 [[main]-pipeline-manager] ERROR logstash.shutdownwatcher - The shutdown process appears to be stalled due to busy or blocked plugins. Check the logs for more information.
11:53:17.168 [[main]-pipeline-manager] FATAL logstash.shutdownwatcher - Forcefully quitting logstash..
11:53:17.241 [[main]-pipeline-manager] ERROR logstash.shutdownwatcher - The shutdown process appears to be stalled due to busy or blocked plugins. Check the logs for more information.
11:53:17.340 [[main]-pipeline-manager] FATAL logstash.shutdownwatcher - Forcefully quitting logstash..
11:53:17.410 [[main]-pipeline-manager] ERROR logstash.shutdownwatcher - The shutdown process appears to be stalled due to busy or blocked plugins. Check the logs for more information.
11:53:17.511 [[main]-pipeline-manager] FATAL logstash.shutdownwatcher - Forcefully quitting logstash..
```

**Comment by original-brownbear** [comment]
@suyograo  this broke yesterday from you removing CEF in e939c6a :) will set those tests to ignore for now.

**Comment by original-brownbear** [comment]
fixed in #7462 :)
